Project1 Completed

peihongx opened the pull request:
## Please review my project 1 and kindly leave your suggestions.

jonathanvajda committed:
Typo fix (#5)

The project was copy-and-pasted, but not all deets were updated to reflect the project is the 2nd one

Jiemingy committed:
Update README.md

Jiemingy committed:
Update README.md

Jiemingy committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Update README.md

johnbeve committed:
Create README.md

johnbeve committed:
Update README.md

Fixed question 8 based on Jieming's note.

johnbeve committed:
Update README.md

You can tag your peers in a commit @jonathanvajda @SydCo99 @giacomodecolle @oliviahobai @Jiemingy @JaronJCheung 

If you want someone to look over your work before you open a pull request, just tag them in the commit.

peihongx committed:
Add proj1

dmcnulty27 commented on Project-1/Peihong_Xie_Project_1_50375781.md:
```diff
@@ -0,0 +1,249 @@
+
+(b) CNF: B∨¬A
+
+(c) CNF: B∨¬B (the CNF of a tautology is any formula of the form "p∨¬p")
```
Could you point me to where you drew this conclusion?

peihongx replied:
Thank you for commenting on my assignment!

Regarding (c), if you run it in a truth table generator like https://web.stanford.edu/class/cs103/tools/truth-table-tool/, you will find it is a tautology because every line of its truth-table is True. 

Regarding (b), you are right that my previous CNF is wrong. But I also find (b) is a tautology, so its CNF is the same as (c).

peihongx committed:
Delete README.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

giacomodecolle commented on Project-1/Peihong_Xie_Project_1_50375781.md:
```diff
@@ -0,0 +1,249 @@
+
+(b) ∃x∀y∃z means “there exists x such that for all y, there exists z such that” while ∀x∃y∀z means "for all x, there exists y such that for all z"
+
+(c) ∀x∃y∀z∃w means "for all x, there exists y such that for all z, there exists w such that" while ∃x∀y∃z∀w means "there exists x such that for all y, there exists z such that for all y"
```
I think you meant "for all w" at the end of line 172

peihongx replied:
> I think you meant "for all w" at the end of line 172

Thanks!

giacomodecolle commented on Project-1/Peihong_Xie_Project_1_50375781.md:
```diff
@@ -0,0 +1,249 @@
+(d) $O ≡ ∃parentOf^-.(∃parentOf\le1)$
+
+(e) $T ≡ (M\sqcap ∃parentOf^-.(\le3∃parentOf.M \sqcap \ge3∃parentOf.M))\sqcup(¬M\sqcap∃parentOf^-.(\le2∃parentOf.M \sqcap \ge2∃parentOf.M))$
+
```
The definitions look like the best thing I got so far on description logic, but could you point out where you found the syntactical rules for applying symbols like "≥2" or the symbols to mean the reverse of a concept? I didn't seem to find them in the parts of the book I was reading

peihongx replied:
> The definitions look like the best thing I got so far on description logic, but could you point out where you found the syntactical rules for applying symbols like "≥2" or the symbols to mean the reverse of a concept? I didn't seem to find them in the parts of the book I was reading

@giacomodecolle My line of thought comes from Baader - An Introduction to Description Logic (one of our reading materials). Please see its Section 2.5. So this means that in order to do this question, we have to read some parts which are beyond our first-week reading. 😭

giacomodecolle replied:
Hi Karl, I was working on this with Ali today and we may have a problem with building the reverse of parentof_, since that is an undefined primitive and we technically have no formal way to give a value to its reverse. Starting from this we built some intermediate concepts which we can use to build the whole thing. Perhaps we can discuss it tomorrow at the meeting, in the meantime here is some tentative notes:

  (a)  B that says that x is a brother of y

  Per (Person) ≡ M ⊔ ¬M
  p2 (Parent of 2) ≡ ∃parentof. ≥ 2 ⊓ ¬(∃parentof. = 1) (this condition is meant to avoid that the two persons are the same, but I am not entirely sure it works or whether there are better ways to do so)
  
  B ≡ M ⊓ ∃p2¯.Per

  (b)  A that says that x is an aunt of y
  
  ∃p2¯.parentof.parentof.Per

  A ≡ ¬M ⊓ (∃p2¯. (parentof. ≥ 2).(parentof.Per)) ⊔ (¬parentof. Per ⊓ ∃p2¯.parentof.parentof.Per)
  
  U(uncle) ≡ M ⊓ (∃p2¯. (parentof. ≥ 2).(parentof.Per)) ⊔ ¬parentof.Per 

  (c)  C that says that x and y are cousins

  gp2(Grandparent with at least two children which each have children) ≡ ∃parentof ≥ 2.parentof.Per

  C ≡ gp2¯.per

  But this would only describe one of the two cousins, i.e. "being a cousin of someone" or "having a cousin".
  We cannot define "being cousins" in description logic.

  (d)  O that says that x is an only child  

  O ≡ ¬ ∃p2¯.Per

  (e)  T that says that x has exactly two brothers 

  This cannot be done without a role name for brother, we only have a concept name for it.

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

peihongx committed:
Update Peihong_Xie_Project_1_50375781.md

Jiemingy commented:
Hi Karl, for your answers to the question 3 (b) and (c), I wonder, if ∧¬(x=y) is needed for (c), ∧¬(x=y) should also be in the answer to (b), since it could be x=y for the code. What do you think?

peihongx commented:
> Hi Karl, for your answers to the question 3 (b) and (c), I wonder, if ∧¬(x=y) is needed for (c), ∧¬(x=y) should also be in the answer to (b), since it could be x=y for the code. What do you think?

@Jiemingy Hi Jieming, thanks for your comment. ¬(x=y) is made to ensure that one is a brother/counsin of oneself. But you are right, if we are careful enough, then we need to follow @Finn1928 and add more non-identity statements: "For question 3, I think we should also add the non-identity statements for the other definitions, not just 3(a). For 3(b) I added x≠y∧x≠z∧x≠w∧w≠y∧w≠z∧y≠z, for 3(c) I added x≠v∧x≠y∧x≠z∧x≠w∧w≠y∧w≠z∧w≠v∧y≠z∧y≠v∧v≠z, for 3(d) I added y≠z∧y≠x, and for 3(e) I added v≠w∧v≠x∧v≠y∧v≠z."

peihongx committed:
Merge branch 'main' into Project1
